app/src/main/java/Fragments/ProfileFragment.java

```diff
@@ -109,18 +120,30 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     public void onStart() {
         super.onStart();
 
-        // TODO: Set listener for add_photo ImageButton to call addPhoto()
-        addPhoto_ib.setOnClickListener(new View.OnClickListener() {
+        updatePhoto_ib.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                editPhoto();
+                updatePhoto();
             }
         });
+        // TODO: Add a single listener for text field modifications
+    }
 
-        // TODO: Set listener for edit_email, edit_phone_number
+    @Override
+    public void onPrepareOptionsMenu(Menu menu) {
+        MenuItem saveMenuItem = menu.findItem(R.id.profile_save_button);
+        saveMenuItem.setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
+            @Override
```

Comment:
If you're not signed in, the save button should be disabled or hidden. Currently it tries to update even if you're signed out.

Reply:
Fixed.